src/client/contexts/CubeContext.tsx

```diff
@@ -240,7 +240,10 @@ export function CubeContextProvider({
   const [sortTertiary, setSortTertiary] = useQueryParam('s3', defaultSorts[2]);
   const [sortQuaternary, setSortQuaternary] = useQueryParam('s4', defaultSorts[3]);
   const [filterResult, setFilterResult] = useState({});
-  const [useBlog, setUseBlog] = useLocalStorage<boolean>(`${cube.id}-useBlog`, true);
+  const [useBlog, setUseBlog] = useLocalStorage<boolean>(
+    `${cube.id}-useBlog`,
+    typeof user?.autoBlog !== 'undefined' ? user.autoBlog : false,
```

Comment:
I could slightly improve the typing by updating the UserContextValue type (in UserContext) to not simply extend User, but define the defaultPrinting, gridTightness, and autoBlog settings as always defined since we are setting them in hydration if they aren't set.
Admittedly all that would do is slightly reduce the verbosity of a statement like this when having to check set or not, which really is only annoying for a boolean compared to a string.